src/dmd/iasm.d

```diff
@@ -4378,7 +4378,9 @@ extern (C++) public Statement asmSemantic(AsmStatement s, Scope *sc)
     if (!s.tokens)
         return null;
 
-    memset(&asmstate, 0, asmstate.sizeof);
+    asmstate.ucItype = 0;
+    asmstate.bReturnax = false;
+    asmstate.lbracketNestCount = 0;
```

Comment:
How about adding a `reset` method to the Asm state struct?

Reply:
Or `asmstate = ASM_STATE.init`?

Reply:
Adding a `reset` member function to do this is a good idea, but I'll approve this even without that change (as a bit more could be done in that direction).